openapi/src/main/java/io/micronaut/openapi/visitor/Utils.java

```diff
@@ -92,14 +95,28 @@ private Utils() {
 
     @Nullable
     public static Path getProjectPath(VisitorContext context) {
-        Path path;
-        try {
-            path = context.getProjectDir().orElse(Utils.isTestMode() ? Paths.get(System.getProperty("user.dir")) : null);
-        } catch (Exception e) {
-            // Should never happen
-            path = Paths.get(System.getProperty("user.dir"));
+
+        Path projectPath = context.get(MICRONAUT_INTERNAL_OPENAPI_PROJECT_DIR, Path.class).orElse(null);
+        if (projectPath != null) {
+            return projectPath;
+        }
+
+        String projectDir = System.getProperty(MICRONAUT_OPENAPI_PROJECT_DIR);
```

Comment:
you shouldn't use system properties but the annotation processor options otherwise the compiler can't detect when input values change

Reply:
I don't quite understand what's wrong here. As I understand it, it is necessary to describe the properties of the processor, which must be allowed to be loaded as system variables. They are described in the `@SupportedOptions` annotation. Now everything has been implemented.

Reply:
in general parameters to an annotation processor should never be ready from system properties and should always be read from annotation processor arguments, this particular processor abuses that badly in several places which is less than ideal.

Reply:
I see, then how is it correct to write the reading of the arguments? Although I'm not sure if there's much to worry about using system variables. I think it's redundant

Reply:
@graemerocher Do I need to write something like this?
```java
context.getOptions().get(MICRONAUT_INTERNAL_OPENAPI_PROJECT_DIR)
```

Reply:
It's just that I don't understand. I give kapt the annotation processor argument, but I read it with a system variable. That is, it turns out that all arguments are already available as system variables. Or does kapt set not an argument, but a system property?

I tested this solution on a test project - it works correctly.

Reply:
yes it needs to be `context.getOptions().get(MICRONAUT_INTERNAL_OPENAPI_PROJECT_DIR)`